### ISSUE TITLE
[VS Component Analytics] [DEFECT] The va-search-input component seemingly does not make use of disable-analytics prop

### ISSUE THREAD

**Issue by FranECross** [department-of-veterans-affairs/va.gov-cms#18846]
## Description

The Sitewide team is working to remove any duplicate Google Analytics events from our products as we pay per event. We have a [ticket ](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/18790)here to resolve this duplicate event issue for Find Forms specifically but we can't complete it until this issue is fixed.

## Issue Blocking our ticket noted above
The va-search-input component seemingly does not make use of disable-analytics prop. When it is added, the basic analytics event for this component still fires:
<img width="943" 

src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBN0l5Qmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--d266deb7653a29e6b7f48152b69f1378ed5f99ad/image.png" alt="image.png" />

You can see this when using the search bar on https://www.va.gov/find-forms, typing a search term, and searching (I use the Chrome extension Adswerve to see the events).

## What I expected to happen
I expected the va-search-input component to allow disabling the analytics. I looked in the source code for this component: https://github.com/department-of-veterans-affairs/component-library/blob/main/packages/web-components/src/components/va-search-input/va-search-input.tsx and the disableAnalytics prop is not present as it is for va-link.

## Reproducing
You can add disable-analytics to the search input in the Find Forms app here: https://github.com/department-of-veterans-affairs/vets-website/blob/b364d52879588928a756ebfcd7ebe7eaf7788aef/src/applications/find-forms/containers/SearchForm.jsx#L97 and run the code locally using yarn watch --env api=https://staging-api.va.gov. Note that the disable-analytics="true" prop shows up in the DOM but the int-search-input-click still fires when the search happens.

## Acceptance criteria

>Consider
> - Design / Accessibility reviews
> - Collab cycle requirements
> - Device sizes (mobile first)
> - Documentation updates / Change management
< - [Content model documentation](https://prod.cms.va.gov/admin/structure/cm_document)
> - Testing notes

- [ ]

**Comment by randimays** [comment]
Pointed at a 2; implementation seems simple but I'm not sure of the code / merge / deploy process on DST; this allows for some ambiguity.

**Comment by randimays** [comment]
Pulling in as stretch for sprint 9

**Comment by randimays** [comment]
@FranECross Our portion of the work for this ticket is complete; the PR has been merged. The code will go out with the next DST release next week. Perhaps we should leave this in prod verification until the release is ready?

**Comment by FranECross** [comment]
> @FranECross Our portion of the work for this ticket is complete; the PR has been merged. The code will go out with the next DST release next week. Perhaps we should leave this in prod verification until the release is ready?

Sounds great, thanks!

**Comment by randimays** [comment]
`component-library` release was made on Friday. We're waiting on [this PR](https://github.com/department-of-veterans-affairs/vets-website/pull/31444) in vets-website to be merged and deployed to production. Once that is complete, we'll need to pull the new code and test locally; we don't have any `va-search-input` components in production with a `disable-analytics` prop on them yet.

**Comment by randimays** [comment]
EOD update: [this PR](https://github.com/department-of-veterans-affairs/vets-website/pull/31444) is still waiting to be merged. Once it has been merged into vets-website, the fix can be validated locally:

In vets-website:
- Do a `yarn install` on `main` to be sure v45 of `component-library` is picked up
- Run `yarn watch --env api=https://staging-api.va.gov` on `main`
- Add `disable-analytics` to [this search input](https://github.com/department-of-veterans-affairs/vets-website/blob/0b5c5214af68d5df173e29fecc639e18e48b3b3f/src/applications/find-forms/containers/SearchForm.jsx#L97) in Find Forms
- Enable your Adswerve Chrome extension and pop open the console
- Go to `/find-forms` and enter a search term in the box and click "Search"
- Validate that you don't see any analytics events firing

If the steps above work as expected, we can close this ticket 👍🏻

**Comment by randimays** [comment]
This is the analytics event that is fired on a `va-search-input` that does not have analytics disabled (screenshot from prod).

<img width="631" alt="Screenshot 2024-09-03 at 4 06 01 PM" src="https://github.com/user-attachments/assets/bdfd1f3f-1343-4d53-b126-452f02ae3063">

I disabled analytics on the Find Forms search input locally and did not see this event fire:

<img width="652" alt="Screenshot 2024-09-03 at 4 06 07 PM" src="https://github.com/user-attachments/assets/8e9fe8bd-39a0-4d68-9aab-7d1bbf0193a3">

Our code changes worked as expected. Closing as complete!
